cmd/zed/zed.d/zed-functions.sh

```diff
@@ -413,3 +413,143 @@ zed_rate_limit()
     zed_unlock "${lockfile}" "${lockfile_fd}"
     return "${rv}"
 }
+
+
+# query_vdev_status (pool, vdev)
+#
+# Given a [pool] and [vdev], return the matching vdev path & status on stdout.
+#
+# Warning: This function does not handle the case of [pool] or [vdev]
+# containing whitespace.  Beware of ShellCheck SC2046.  Caveat emptor.
+#
+# Arguments
+#   pool: pool name
+#   vdev: virtual device name
+#
+# StdOut
+#   arg1: vdev pathname
+#   arg2: vdev status
+#
+query_vdev_status()
+{
+    local pool="$1"
+    local vdev="$2"
+    local t
+
+    vdev="$(basename -- "${vdev}")"
+    ([ -n "${pool}" ] && [ -n "${vdev}" ]) || return
+    t="$(printf '\t')"
+
+    "${ZPOOL}" status "${pool}" 2>/dev/null | sed -n -e \
+        "s,^[ $t]*\(.*${vdev}\(-part[0-9]\+\)\?\)[ $t]*\([A-Z]\+\).*,\1 \3,p" \
+        | tail -1
+}
+
+
+# get_dev_path (device)
+#
+# Find the exact path of a block device (or link)
+#
+# Arguments
+#   dev: Device name, such as 'sda'
+#
+# Return:
+#   Full path to the device node or link
+get_dev_path()
+{
+    find /dev -type b -name "${1}" -o -type l -name "${1}" | \
+	tr -d "\n"
+}
```

Comment:
Here, possibly check if the device we found is a link, and if so, trace it back to it's "physical" device node. ?